kyo-kernel/shared/src/main/scala/kyo/kernel/Pending.scala

```diff
@@ -435,40 +435,32 @@ object `<`:
     end abortCastUnitImpl
 
     /** Converts a pure single-argument function to an effectful computation. */
-    implicit inline def liftPureFunction1[A1, B](inline f: A1 => B)(
-        using inline flat: CanLift[B]
-    ): A1 => B < Any =
-        a1 => f(a1)
+    implicit inline def liftPureFunction1[A1, B](inline f: A1 => B)(using inline canLift: CanLift[B]): A1 => B < Any = f.asInstanceOf
```

Comment:
This is incorrect. Even if `B` isn't statically known as a Kyo computation via `CanLift`, it may still be a computation in generic scopes and we need to ensure the `Nested` wrapper is instantiated by reflecting at runtime

Reply:
@fwbrasil ok, can we make a test for that? there is a way to optimize it (same as lift).

Reply:
This example doesn't run, even in `main`.

```scala
        "nested effect suspension as function" in {
            def f(str:String): Int < TestEffect2 = TestEffect2(str)
            
            def g[B:WeakFlat](f: String => B):B < TestEffect1 = { 
                val g: String => B < Any = f
                TestEffect1(1).map(g)
            } //(<.liftPureFunction1(f))
            
            inline def h[B](f: String => B): B < TestEffect1 = g(f)
            
            val nested: Int < TestEffect2 < TestEffect1 = h(f)

            val result = TestEffect1.run(nested.map(TestEffect2.run))
            assert(result.eval == 19)

            val result2 = TestEffect1.run(TestEffect2.run(nested.flatten))
            assert(result2.eval == 19)
        }

```

Reply:
do you mean it doesn't compile?  I think you'd need to remove the `WeakFlat` constraint from `g` and also not handle both effects before calling `eval`. That should trigger this branch: https://github.com/getkyo/kyo/blob/803e8420a8563500e39b3e7fdbf708af1b48fd06/kyo-kernel/shared/src/main/scala/kyo/kernel/Pending.scala#L389

Reply:
`h`was not necessary, so the test on `main` is:
```scala
"nested effect suspension as function" in {
            def f(str:String): Int < TestEffect2 = TestEffect2(str)
            
            def g[B](f: String => B):B < TestEffect1 = { 
                val liftedF: String => B < Any = f
                TestEffect1(1).map(liftedF)
            }
            
            val nested: Int < TestEffect2 < TestEffect1 = g(f)

            val result = TestEffect1.run(nested.map(TestEffect2.run))
            assert(result.eval == 19)

            val result2 = TestEffect1.run(TestEffect2.run(nested.flatten))
            assert(result2.eval == 19)
        }
```

Reply:
does it pass or fail on main?